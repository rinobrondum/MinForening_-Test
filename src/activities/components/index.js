export { default as ActivitiesList } from './ActivitiesList'
