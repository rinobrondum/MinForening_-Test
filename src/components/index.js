export {default as Button} from './Button'
export {default as DisabledButton} from './DisabledButton'
export {default as InputWithButton} from './InputWithButton'
export {default as LinkButton} from './LinkButton'
export {default as Hr} from './Hr'
export {default as Searchable} from './Searchable'
export {default as Foldable} from './Foldable'
export {default as Switch} from './Switch'
export {default as LabeledSelect} from './LabeledSelect'
export {default as LabeledTextArea} from './LabeledTextArea'
export {default as ButtonWithProtectedAction} from './ButtonWithProtectedAction'
export {default as DownloadTheApp} from './DownloadTheApp'
export {default as LabeledDateInput} from './LabeledDateInput'
export {default as LabeledSwitch} from './LabeledSwitch'
export {default as ConfirmModal} from './ConfirmModal'
export {default as FieldWithLabel} from './FieldWithLabel'
export {default as Card} from './Card'
export {default as Page} from './Page'
export {default as Dropdown} from './Dropdown'
export {default as OpenableImage} from './OpenableImage'
export {default as Link} from './Link'
export {default as Input} from './Input'
export {default as CopyToClipboardButton} from './CopyToClipboardButton'
export {default as Text} from './Text'
export {default as Dropzone} from './Dropzone'
export {default as Modal} from './Modal'
export {default as Select} from './Select'
export {default as Image} from './Image'
export {default as TextArea} from './TextArea'
export {default as LabeledInput} from './LabeledInput'
export {default as Checkbox} from './Checkbox'
export {default as DateInput} from './DateInput'
export {default as Loading} from './Loading'
export {default as H1} from './H1'
export {default as H2} from './H2'
export {default as CloseButton} from './CloseButton'
export {default as NotificationBadge} from './NotificationBadge'
export {default as DialogueModal} from './DialogueModal'
export {default as LabeledCheckbox} from './LabeledCheckbox'
export {default as StyledCheckbox} from './StyledCheckbox'
export {default as StyledRadioButton} from './StyledRadioButton'
export {default as FormikInput} from './FormikInput'
export {default as Box} from './Box'
export {default as Flex} from './Flex'
export {default as ImageInput} from './ImageInput'
export {default as Tooltip} from './Tooltip'
export {default as ClubCard} from './ClubCard'
export {default as PaymentMethodList} from './PaymentMethods'
