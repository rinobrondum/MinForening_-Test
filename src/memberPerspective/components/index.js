export {default as MemberPerspective} from './MemberPerspective'
export {default as MemberChat} from './MemberChat'
