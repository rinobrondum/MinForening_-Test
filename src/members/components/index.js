export { default as SearchableMemberList } from './SearchableMemberList'
export { default as MemberList } from './MemberList'
export { default as MemberTable } from './MemberTable'
