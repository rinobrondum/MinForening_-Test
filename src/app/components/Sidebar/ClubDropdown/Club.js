import React from 'react'
import {Text, Image, Flex} from 'components'

const Club = ({bold = false, clubName: name, imageUrl: image}) => (
  <Flex alignItems="center">
    {image && <Image round mr={2} width="25" height="25" src={image} />}
    <Text light bold={bold} truncate>
      {name}
    </Text>
  </Flex>
)

export default Club
