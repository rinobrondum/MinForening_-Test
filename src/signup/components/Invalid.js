import React from 'react'
import {Flex, Box} from '@rebass/grid'
import {connect} from 'react-redux'
import {createStructuredSelector} from 'reselect'
import useCustomTranslation from 'lib/customT'
import TransWrapper from 'lib/transWrapper'
import {DownloadTheApp, Text} from 'components'
import { getAppName } from 'app/selectors';
const Invalid = ({appName, whiteLabelData}) => {
  const t = useCustomTranslation()

  return (
    <Flex flexDirection="column" alignItems="center">
      <Box mb={4}>
        <Text bg="danger" light center>
          <strong>{t('invitationLinkVoid')}</strong>
        </Text>
      </Box>

      <Box style={{maxWidth: 350}}>
        <Box mb={4}>
          <Text center>
            <TransWrapper i18nKey="contactVoidInvitationSender">
              Tag kontakt til personen du modtog linket af, og se om du kan f√•
              et <strong>gyldigt invitationslink</strong> til din forening.
            </TransWrapper>
          </Text>
        </Box>
        <Box mb={3}>
          <Text center>
            <TransWrapper i18nKey="theAppCanBeDownloaded">
              {{appName}} App kan downloades via <strong>Google Play</strong>{' '}
              eller <strong>App Store</strong>
            </TransWrapper>
          </Text>
        </Box>

        <DownloadTheApp whiteLabelData={whiteLabelData} />
      </Box>
    </Flex>
  )
}
const enhancer = connect(createStructuredSelector({appName: getAppName}))

export default enhancer(Invalid)
