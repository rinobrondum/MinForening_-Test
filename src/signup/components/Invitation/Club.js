import React from 'react'
import {Text, Image, Box, H2} from 'components'

const Club = ({name, groups = [], image}) => (
  <Box bg="secondary" p={4}>
    {image && (
      <Image
        round
        src={image}
        width="65"
        height="65"
        mx="auto"
        mb={2}
        display="block"
      />
    )}

    <H2 color="white" textAlign="center">
      <strong>{name}</strong>
    </H2>

    {groups.length > 0 && (
      <Text color="white" textAlign="center" mt={2}>
        {groups.join(', ')}
      </Text>
    )}
  </Box>
)

export default Club
