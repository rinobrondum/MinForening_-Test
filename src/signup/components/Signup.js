import React, {useEffect, useState} from 'react'
import {connect} from 'react-redux'
import {compose} from 'recompose'
import {createStructuredSelector} from 'reselect'
import {Switch, Route, Redirect} from 'react-router-dom'
import {withRouterParams} from 'lib/hoc'
import {getIsAuthenticated} from 'signup/user'
import {fetchDummies} from 'signup/actions'
import {Page, Box} from 'components'
import {LogoWithTagline} from 'components/icons'
import UserList from './UserList'
import User from './User'
import Invalid from './Invalid'
import Done from './Done'
import { globalState } from 'app/components/App'

export const Signup = ({
  fetchDummies,
  match: {path},
  params: {email, password},
  location: {pathname},
}) => {
  const [invalid, setInvalid] = useState(!email && !password)
  const [ready, setReady] = useState(false)
  const [whiteLabelData, setWhiteLabelData] = useState(null)
  
  useEffect(() => {

    new Promise(async (resolve, reject) => {
      if (!whiteLabelData) {
        await fetch('/paths/paths.json?q=' + Math.floor(Math.random() * 10000000).toString())
          .then((res) => res.json())
          .then((data) => {
            setWhiteLabelData(data);
            globalState.setState('apiMyOrgUrl', data.appSettings.apiMyOrgUrl);
            globalState.setState('domainSettings', data.domainSettings);
        });
      }
     
      if (whiteLabelData) {
        if (email && password) {
          return fetchDummies({email, password, resolve, reject})
        } else {
          setReady(true)
          setInvalid(true)
        }
      }
    })
      .then(() => setReady(true))
  }, [email, password, fetchDummies, whiteLabelData])
  
  return ready ? (
    invalid && pathname !== '/register/invalid' ? (
      <Redirect to="/register/invalid" />
    ) : (
      <>
        {whiteLabelData &&
          <Page>
            <Box my={4}>
              {
                whiteLabelData && whiteLabelData.logoTexts && 
                  <LogoWithTagline height="50px" width="100%" whiteLabelData={whiteLabelData}/>
              }
            </Box>

            <Switch>
              <Route exact path={path} component={UserList} />
              <Route exact path={`${path}/done`} render={(props) => <Done {...props} whiteLabelData={whiteLabelData}/>} />
              <Route exact path={`${path}/invalid`} render={(props) => <Invalid {...props} whiteLabelData={whiteLabelData}/>} />
              <Route path={`${path}/:id`} component={User} />
            </Switch>
          </Page>
        }
      </>
    )
  ) : null
}

const enhancer = compose(
  withRouterParams,
  connect(
    createStructuredSelector({
      isAuthenticated: getIsAuthenticated,
    }),
    {
      fetchDummies: fetchDummies.requested,
    }
  )
)

export default enhancer(Signup)
