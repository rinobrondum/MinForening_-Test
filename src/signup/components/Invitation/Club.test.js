import React from 'react'
import { shallow } from 'enzyme'
import { Image, H2, Text } from 'components'
import Club from './Club'

function setup(override) {
  const props = {
    image: 'test',
    name: 'Club name',
    groups: ['First', 'Second'],
    ...override,
  }

  const wrapper = shallow(<Club {...props} />)

  return {
    props,
    wrapper,
  }
}

describe('<Club />', () => {
  it('renders without crashing', () => {
    const { wrapper } = setup()

    expect(wrapper).toMatchSnapshot()
  })

  it('renders image if present', () => {
    const { wrapper } = setup()

    expect(wrapper.find(Image).exists()).toBe(true)
  })

  it('renders club name as header', () => {
    const {
      wrapper,
      props: { name },
    } = setup()

    expect(wrapper.find(H2).exists()).toBe(true)
    expect(
      wrapper
        .find(H2)
        .children()
        .text()
    ).toEqual(name)
  })

  it('renders groups concatenated', () => {
    const {
      wrapper,
      props: { groups },
    } = setup()

    expect(wrapper.find(Text).text()).toEqual(groups.join(', '))
  })
})
