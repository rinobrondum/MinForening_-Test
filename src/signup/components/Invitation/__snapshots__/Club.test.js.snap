// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Club /> renders without crashing 1`] = `ShallowWrapper {}`;
